clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/config/KubernetesConfigurationProperties.java

```diff
@@ -75,4 +85,93 @@ public void validate() {
+    boolean accountExist = false, permissionsExist = false, crdExist = false;
+    Pattern iPattern = Pattern.compile(propertyRegex);
+    Matcher match;
+    String propertyName;
+    int currentAccIndex = 0, tempPropertyIndex;
+    int currentCrdIndex = 0, tempCrdIndex = 0;
+    Object value;
+    for (Map.Entry<String, Object> entry : originalPropertiesMap.entrySet()) {
+      value = entry.getValue();
+      match = iPattern.matcher(entry.getKey());
+      if (match.matches() && value != null && value != "") {
+        tempPropertyIndex = Integer.parseInt(match.group(1));
+        propertyName = match.group(2);
+        if (tempPropertyIndex != currentAccIndex) {
+          // settle previous account
+          wrapper.setPropertyValues(accountPropertiesMap);
+          // setup for new Account
+          currentAccIndex = tempPropertyIndex;
+          accountExist = false;
+          permissionsExist = false;
+          currentCrdIndex = 0;
+          crdExist = false;
+        }
+        if (!accountExist) {
+          wrapper = new BeanWrapperImpl(ManagedAccount.class);
+          getAccounts().add((ManagedAccount) wrapper.getWrappedInstance());
+          accountExist = true;
+          accountPropertiesMap = new HashMap<>();
+        }
+        if (propertyName.equals("permissions")) {
+          if (!permissionsExist) {
+            accountPermissions = new Permissions.Builder();
+            accountPropertiesMap.put("permissions", accountPermissions);
+            permissionsExist = true;
+          }
+          accountPermissions
+              .computeIfAbsent(Authorization.valueOf(match.group(6)), a -> new ArrayList<>())
+              .add((String) value); // match.group(6) is Authorization string
+        } else if (propertyName.equals("customResources")) {
+          tempCrdIndex = Integer.parseInt(match.group(4)); // match.group(4) is the index of crd
+          if (currentCrdIndex != tempCrdIndex) {
+            currentCrdIndex = tempCrdIndex;
+            crdExist = false;
+          }
+          if (!crdExist) {
+            crdWrapper = new BeanWrapperImpl(CustomKubernetesResource.class);
+            accountPropertiesMap.put(
+                "customResources[" + currentCrdIndex + "]", crdWrapper.getWrappedInstance());
+            crdExist = true;
+          }
+          crdWrapper.setPropertyValue(
+              match.group(6),
+              entry.getValue()); // match.group(6) is a property of CustomKubernetesResource
+        } else {
+          if (propertyName.equals("kubeconfigFile")) {
+            if (((String) value).startsWith("configserver:")) {
+              value = propertiesMapExtractor.resolveConfigServerFilePath((String) value);
+            }
+          }
+          accountPropertiesMap.put(
+              entry.getKey().substring(entry.getKey().indexOf(propertyName)), value);
+        }
+      } else {
+        log.debug("Ignoring the properties having blank values. Key:" + entry.getKey());
```

Comment:
Seems that this code ignores all properties not belonging to accounts. What happens for properties at the `kubernetes` top level like `rawResourcesEndpointConfig`?

Reply:
Now rawResourcesEndpointConfig is included.